airqo-platform/AirQo-api
# Enhancements to the API product (V2)

**_WHAT DOES THIS PR DO?_**

- [x] installed [moesif package](https://www.moesif.com/) to log API calls and send to dashboard for analytics
- [x] Resolve an issue regarding querying by datetime. Please note that timestamp defaults to UTC iF the timezone is not specified in your provided datetime.
- [x] ability to query by device_number, order output by timestamp
- [x] Sanitises timestamp output format, Initially, the timestamp returned as an object instead of just the string value but it is now just a String.
- [x] The returned measurements are now returned in descending order with the most recent one on top. 

**_WHAT ISSUES ARE RELATED TO THIS PR?_**

- Jira cards
    - [PLAT-1197]

**_HOW DO I TEST OUT THIS PR?_**
```
cd src/device-registry
npm run install
npm run prod-mac
```
Run instructions different based on operating system, please check README
Also ensure that you update your **.env** file with this new variable's value: `MOESIF_APPLICATION_ID`

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 

- [x] [Get measurements V2](https://app.gitbook.com/o/-MCogOVGQqjUharJKchL/s/-MKqyQkcfs54GYi-q96G/device-registry/events#version-2-get-events)

**_SCREENSHOTS?:_** 
<img width="834" alt="Screenshot 2022-07-26 at 20 53 18" src="https://user-images.githubusercontent.com/1590213/181077607-8f7f0dcf-00de-44c9-a99a-313bea088730.png">




[PLAT-1197]: https://airqoteam.atlassian.net/browse/PLAT-1197?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
